### PR TITLE
Do not override enchantments of the `stored_enchantments` component

### LINKED ISSUE #887: [1.20.5] `minecraft:stored_enchantments` item subpredicate does not function
**Minecraft Version:** 1.20.5

**NeoForge Version:** 20.5.20-beta

**Logs:** https://mclo.gs/bjV9DXR

**Steps to Reproduce:**
 1. [Download this datapack](http://img.merchantpug.net/u/AdfkGs.zip).
 2. Put it in your world folder.
 3. Give yourself an Aqua Affinity enchanted book.
 4. Run the command `/item modify entity @s weapon.mainhand test:oops`.
 5. It does not work on NeoForge, but works on vanilla.

**Description of issue:**
The `minecraft:stored_enchantments` item subpredicate does not function properly. This is because there is no component check to make sure that the enchantments specified are base enchantments before applying the NeoForge enchantment changes.

Vanilla Instance:
image

NeoForge Instance:
image

### PR DESCRIPTION
**Pull request by Matyrobbrt**
Fixes #887

### CODE REVIEW

File: patches/net/minecraft/FileUtil.java.patch [hunk 1]
```diff
@@ -27,7 +27,7 @@
 @@ -74,7 +_,7 @@
  
      public static boolean isPathPortable(Path p_133735_) {
-         for(Path path : p_133735_) {
+         for (Path path : p_133735_) {
 -            if (RESERVED_WINDOWS_FILENAMES.matcher(path.toString()).matches()) {
 +            if ((net.neoforged.neoforge.common.NeoForgeMod.getProperFilenameValidation() ? RESERVED_WINDOWS_FILENAMES_NEOFORGE : RESERVED_WINDOWS_FILENAMES).matcher(path.toString()).matches()) {
```
Comment (FiniteReality) [inline patches/net/minecraft/FileUtil.java.patch:32]: Is this necessary for this change? It seems unrelated to me.

Comment (TelepathicGrunt) [reply patches/net/minecraft/FileUtil.java.patch:32]: Wondering if our IDEs keep adding the space. It is showing up for me now when I do unpack sources. My original pr to this area did not have the space
https://github.com/neoforged/NeoForge/pull/767/files

Comment (FiniteReality) [reply patches/net/minecraft/FileUtil.java.patch:32]: Oh, I misread the diff, silly me. Yeah, it seems like this was an extraneous whitespace change, which we can ignore or just merge to stop it from happening again.

File: patches/net/minecraft/advancements/critereon/ItemEnchantmentsPredicate.java.patch [hunk 2]
```diff
@@ -1,14 +1,14 @@
...
-     }
+@@ -48,6 +_,11 @@
+             super(p_333967_);
+         }
  
-+    @Override // NEO: use getAllEnchantments for enchantments
-+    public boolean matches(ItemStack p_333958_) {
-+        return matches(p_333958_, p_333958_.getAllEnchantments());
-+    }
++        @Override // NEO: use getAllEnchantments for enchantments
```
Comment (Shadows-of-Fire) [inline patches/net/minecraft/advancements/critereon/ItemEnchantmentsPredicate.java.patch]: Should be `Neo`

**Comment by embeddedt** [comment]
What is the actual change in this PR?

**Comment by Matyrobbrt** [comment]
> What is the actual change in this PR?

The patch is moved from the superclass to the actual enchantments subclass.

**Review by embeddedt — APPROVED** [review]

**Pull request merged**